migrations/versions/66b67130c901_retroactively_migrate_previous_id_for_.py

```diff
@@ -0,0 +1,109 @@
+"""retroactively_migrate_previous_id_for_line_item
+
+Revision ID: 66b67130c901
+Revises: 171fcb171759
+Create Date: 2017-10-26 14:50:18.859247
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '66b67130c901'
+down_revision = '171fcb171759'
+
+from collections import OrderedDict
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.sql import table, column
+import sqlalchemy_utils
+from boxoffice.models.order import ORDER_STATUS
+from boxoffice.models.line_item import LINE_ITEM_STATUS
+
+
+
+def find_nearest_timestamp(lst, timestamp):
+    nearest_ts = None
+    for ts in lst:
+        diff = abs(timestamp - ts).total_seconds()
+        if diff < 1:
+            if nearest_ts and diff > abs(timestamp - nearest_ts).total_seconds():
+                # there's a timestamp that's closer
+                pass
+            else:
+                nearest_ts = ts
+    return nearest_ts
```

Comment:
This could be simplified to ([source](https://stackoverflow.com/a/12141207/78903)):

```python
nearest_ts = min(lst, key=lambda x: abs(x - timestamp).total_seconds())
if abs(nearest_ts - timestamp).total_seconds() < 1:
    return nearest_ts
```